src/documentation/developers/grid_deployment/app_case_study_nextcloud.md

```diff
@@ -0,0 +1,287 @@
+  - [Add an Icon](#add-an-icon)
+  - [Add a Description](#add-a-description)
+  - [Update deployment\_list.ts](#update-deployment_listts)
+  - [Update /types/index.ts](#update-typesindexts)
+  - [Update /router/index.ts](#update-routerindexts)
+  - [Update vm\_deployment\_table.vue](#update-vm_deployment_tablevue)
+  - [Update delete\_deployment.ts](#update-delete_deploymentts)
+  - [Update tf\_deployment\_list.vue](#update-tf_deployment_listvue)
+  - [Create nextcloud\_view.vue](#create-nextcloud_viewvue)
+  - [Create tf\_nextcloud.vue](#create-tf_nextcloudvue)
+    - [Template Section](#template-section)
+    - [First Script Section (Setup)](#first-script-section-setup)
+    - [Second Script Section](#second-script-section)
+- [Testing a application](#testing-a-application)
+- [Contributing to the Dashboard](#contributing-to-the-dashboard)
+- [Conclusion](#conclusion)
+
+***
+
+# Introduction
+
+In this ThreeFold guide, we explore how to create an application on the Dashboard. To do so, we will use the Nextcloud application as a case study. This application is ideal to study since it uses many of the ThreeFold Grid functionalities, such as gateways and gateway domains.
+
+By reading this guide, you should have a proper understanding of the general process of building a new application on the ThreeFold Grid. This might give you inspiration to build your own application and contribute to the ThreeFold Dashboard.
+
+# Process Overview and Preparation Steps
+
+The bulk of the files needed for the Nextcloud application can be found in the [ThreeFold tfgrid-sdk-ts repository](https://github.com/threefoldtech/tfgrid-sdk-ts) on GitHub. More specifically, we are interested in the subdirectory called [playground](https://github.com/threefoldtech/tfgrid-sdk-ts/tree/development/packages/playground).
+
+While there are many ways to proceed in the development of a ThreeFold application for the Dashboard, we present here a general method that works efficiently and is well organized.
+
+The information provided here are specific to the Nextcloud workload, but it can be applied to other types of workload.
+
+Before building the Nextcloud application, we first deployed a [Nextcloud instance](../../system_administrators/terraform/advanced/terraform_nextcloud_aio.md) on the ThreeFold Grid with a full virtual machine. Once this deployment was working well, we built an [Nextcloud flist](../flist/flist_case_studies/flist_nextcloud_case_study.md). You can consult the Nextcloud flist code in the ThreeFold Tech [tf-images repository](https://github.com/threefoldtech/tf-images/tree/development/tfgrid3/nextcloud). We note that the flist uses a micro VM. There are some differences between a full VM and a micro VM. We propose users to first start deploying with a full VM and then adjust their work when they want to publish an flist with a micro VM, since full VM are easier to work with. You can of course start directly with a micro VM if you want.
```

Comment:
```suggestion
Before building the Nextcloud application, we first deployed a [Nextcloud instance](../../system_administrators/terraform/advanced/terraform_nextcloud_aio.md) on the ThreeFold Grid with a full virtual machine. Once this deployment was working well, we built a [Nextcloud flist](../flist/flist_case_studies/flist_nextcloud_case_study.md). You can consult the Nextcloud flist code in the ThreeFold Tech [tf-images repository](https://github.com/threefoldtech/tf-images/tree/development/tfgrid3/nextcloud). We note that the flist uses a micro VM. There are some differences between a full VM and a micro VM. We propose users to first start deploying with a full VM and then adjust their work when they want to publish a flist with a micro VM, since full VM are easier to work with. You can of course start directly with a micro VM if you want.
```